pkg/gui/commits_panel.go

```diff
@@ -448,3 +454,63 @@ func (gui *Gui) handleSwitchToCommitFilesPanel(g *gocui.Gui, v *gocui.View) erro
 
 	return gui.switchFocus(g, v, gui.getCommitFilesView())
 }
+
+func (gui *Gui) handleToggleDiffCommit(g *gocui.Gui, v *gocui.View) error {
+	selectLimit := 2
+
+	// get selected commit
+	commit := gui.getSelectedCommit(g)
+	if commit == nil {
+		return gui.renderString(g, "main", gui.Tr.SLocalize("NoCommitsThisBranch"))
+	}
+
+	// if already selected commit delete
+	if idx, has := gui.hasCommit(gui.State.DiffEntries, commit.Sha); has {
+		gui.State.DiffEntries = gui.deleteCommit(gui.State.DiffEntries, idx)
+	} else {
+		if len(gui.State.DiffEntries) == selectLimit {
+			gui.State.DiffEntries = gui.deleteCommit(gui.State.DiffEntries, 0)
+		}
+		gui.State.DiffEntries = append(gui.State.DiffEntries, commit)
+		gui.setDiffMode()
+	}
+
+	// if selected two commits, display diff between
+	if len(gui.State.DiffEntries) == selectLimit {
+		commitText, err := gui.GitCommand.DiffCommits(gui.State.DiffEntries[0].Sha, gui.State.DiffEntries[1].Sha)
+
+		if err != nil {
+			return gui.createErrorPanel(gui.g, err.Error())
+		}
+
+		return gui.renderString(g, "main", commitText)
+	}
+
+	return nil
+}
+
+func (gui *Gui) setDiffMode() {
+	v := gui.getCommitsView()
+	if len(gui.State.DiffEntries) != 0 {
+		gui.State.Panels.Commits.SpecificDiffMode = true
+		v.Title = gui.Tr.SLocalize("CommitsDiffTitle")
+	} else {
+		gui.State.Panels.Commits.SpecificDiffMode = false
+		v.Title = gui.Tr.SLocalize("CommitsTitle")
+	}
+
+	gui.refreshCommits(gui.g)
+}
+
+func (gui *Gui) hasCommit(commits []*commands.Commit, target string) (int, bool) {
+	for idx, commit := range commits {
+		if commit.Sha == target {
+			return idx, true
+		}
+	}
+	return -1, false
+}
+
+func (gui *Gui) deleteCommit(commits []*commands.Commit, i int) []*commands.Commit {
```

Comment:
at first pass, this function's name suggests we are actually deleting a commit, but we're really just removing it from our array of DiffEntries. I think we should rethink the naming here.

Perhaps instead of DiffEntries we call them ChosenCommits (can't use SelectedCommits because we use selected to describe the current highlighted commit). And then here we could call the method 'RemoveFromChosenCommits' or 'UnchooseCommit'

Reply:
I see, how about UnchooseDiffEntry?

Reply:
I don't want to use DiffEntry in the name because we may in the future end up using the concept of a 'chosen commit' for some other context like picking commits while doing a git bisect. So 'UnchooseCommit' is a better option I believe

Reply:
OK :)